client/src/main/scala/cakesolutions/kafka/KafkaProducer.scala

```diff
@@ -149,7 +151,13 @@ final class KafkaProducer[K, V](val producer: JKafkaProducer[K, V]) {
     */
   def send(record: ProducerRecord[K, V]): Future[RecordMetadata] = {
     val promise = Promise[RecordMetadata]()
-    producer.send(record, producerCallback(promise))
+    Try {
```

Comment:
Could you use Scala's `try { .. } catch { .. }` expression here instead of the `Try` value? The function only uses the created `Try` value for matching locally, which can also be achieve using try-catch.